### PR TITLE
[New Rule] c89/c99 shell evasion threat

### PR DESCRIPTION
**Pull request by shashank-elastic**
## Issues
https://github.com/elastic/detection-rules/issues/1839

## Summary
c89 an IOC, the Unix binary can be abused to breakout out of restricted shells or environments by spawning an interactive system shell. This activity is not standard use with this binary for a user or system administrator. It indicates a potentially malicious actor attempting to improve the capabilities or stability of their access.

## Contributor checklist

- Have you signed the [contributor license agreement](https://www.elastic.co/contributor-agreement)?
- Have you followed the [contributor guidelines](https://github.com/elastic/detection-rules/blob/main/CONTRIBUTING.md)?

### CODE REVIEW

**Review by DefSecSentinel — APPROVED** [review]
LGTM

File: rules/linux/defense_evasion_c89_binary.toml [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+creation_date = "2022/03/15"
+maturity = "production"
+updated_date = "2022/03/15"
+
+[rule]
+author = ["Elastic"]
+description = """
+Identifies Linux binary c89/c99 abuse to break out from restricted environments by spawning an interactive system
+shell.This activity is not standard use with this binary for a user or system administrator. It indicates a potentially
+malicious actor attempting to improve the capabilities or stability of their access
```
Comment (w0rk3r) [inline rules/linux/defense_evasion_c89_binary.toml:11]: Maybe we can elaborate a less generic description, describe what the binary does, etc

File: rules/linux/defense_evasion_c89_binary.toml [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+"""
+from = "now-9m"
+index = ["logs-endpoint.events.*"]
+language = "eql"
+license = "Elastic License v2"
+name = "Linux Restricted Shell Breakout via c89/c99 Shell evasion"
```
Comment (w0rk3r) [inline rules/linux/defense_evasion_c89_binary.toml:17]:
```suggestion
name = "Restricted Shell Escape via c89/c99"
```

File: rules/linux/defense_evasion_c89_binary.toml [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+rule_id = "1859ce38-6a50-422b-a5e8-636e231ea0cd"
+severity = "medium"
+tags = ["Elastic", "Host", "Linux", "Threat Detection", "Defense Evasion", "GTFOBins"]
+timestamp_override = "event.ingested"
+type = "eql"
+
+query = '''
+sequence by host.id, process.pid with maxspan=1m
+[process where process.name :("c89","c99") and process.args == "-wrapper" and process.args : ("sh,-s", "bash,-s", "dash,-s", "/bin/sh,-s", "/bin/bash,-s", "/bin/dash,-s")]
+[process where process.parent.name : ("bash", "sh", "dash")]
```
Comment (w0rk3r) [inline rules/linux/defense_evasion_c89_binary.toml:29]:
```suggestion
[process where process.name : ("c89","c99") and process.args : "-wrapper" and
   (process.args : ("sh", "bash", "dash") and
   process.args : "-s")]
[process where process.parent.name : ("bash", "sh", "dash")]
```

We can discuss this one, but looks like `sh,-s` args could be bypassed with spaces, would `sh, -s` work?

**Review by w0rk3r — CHANGES_REQUESTED** [review]
Cool one, left a few comments for us to discuss

File: rules/linux/defense_evasion_c89_binary.toml [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+[[rule.threat]]
+framework = "MITRE ATT&CK"
+[[rule.threat.technique]]
+id = "T1548"
+name = "Abuse Elevation Control Mechanism"
+reference = "https://attack.mitre.org/techniques/T1548/"
+[[rule.threat.technique.subtechnique]]
+id = "T1548.004"
+name = "Elevated Execution with Prompt"
+reference = "https://attack.mitre.org/techniques/T1548/004/"
```
Comment (w0rk3r) [inline rules/linux/defense_evasion_c89_binary.toml:42]:
```suggestion
```
I don't think this technique fits here, if the execution of the command does not prompt the user for credentials

File: rules/linux/defense_evasion_c89_binary.toml (hunk 1 above)
Comment (shashank-elastic) [reply rules/linux/defense_evasion_c89_binary.toml:11]: The description was more like a template for all the detections going out in terms of shell evasion for UNIX libraries. 
What could be captured in one more sentence is "The c89 utility is an interface to the standard C compilation system and it is uncommon for any user to spawn a shell from this utility" 

Can be paraphrased better and taken in code sanity before feature merge for all rules description.

File: rules/linux/defense_evasion_c89_binary.toml (hunk 4 above)
Comment (DefSecSentinel) [reply rules/linux/defense_evasion_c89_binary.toml:42]: I agree. This isn't privilege escalation.

**Review by Samirbous — APPROVED** [review]

File: rules/linux/defense_evasion_c89_binary.toml (hunk 1 above)
Comment (DefSecSentinel) [reply rules/linux/defense_evasion_c89_binary.toml:11]: Yeah we may want to go back in and add another sentence to these rules defining the binary and it's normal usage.

**Review by w0rk3r — APPROVED** [review]

**Pull request merged**